Buildings/Fluid/HeatExchangers/ThermalWheels/Latent/BaseClasses/Effectiveness.mo

```diff
@@ -0,0 +1,163 @@
+  parameter Modelica.Units.SI.Efficiency epsLatHeaPL(final max=1)
+    "Part load (75%) latent heat exchanger effectiveness at the heating mode";
+  parameter Modelica.Units.SI.VolumeFlowRate VSup_flow_nominal
+    "Nominal supply air flow rate";
+
+  Modelica.Blocks.Interfaces.RealInput TSup(
+    final min=0,
+    final unit="K",
+    displayUnit="degC")
+    "Supply air temperature"
+    annotation (Placement(transformation(extent={{-140,-60},{-100,-20}})));
+  Modelica.Blocks.Interfaces.RealInput TExh(
+    final min=0,
+    final unit="K",
+    displayUnit="degC")
+    "Exhaust air temperature
+    " annotation (Placement(transformation(extent={{-140,-100},{-100,-60}})));
+  Modelica.Blocks.Interfaces.RealInput VSup_flow(final unit="m3/s")
+    "Supply air volumetric flow rate"
+    annotation (Placement(transformation(extent={{-140,60},{-100,100}})));
+  Modelica.Blocks.Interfaces.RealInput VExh_flow( final unit="m3/s")
+    "Exhaust air volumetric flow rate"
+    annotation (Placement(transformation(extent={{-140,20},{-100,60}})));
+  Modelica.Blocks.Interfaces.RealInput uSpe(
+    final unit="1",
+    final max=1)
+    "Wheel speed ratio"
+    annotation (Placement(transformation(extent={{-140,-20},{-100,20}})));
+  Modelica.Blocks.Interfaces.RealOutput epsSen(final unit="1")
+    "Sensible heat exchanger effectiveness"
+    annotation (Placement(transformation(extent={{100,30},{120,50}}),
+        iconTransformation(extent={{100,20},{120,40}})));
+  Modelica.Blocks.Interfaces.RealOutput epsLat(final unit="1")
+    "Latent heat exchanger effectiveness"
+    annotation (Placement(transformation(extent={{100,-50},{120,-30}}),
+        iconTransformation(extent={{100,-40},{120,-20}})));
+
+protected
+  Real rat
+    "Ratio of the average operating air flow rate to the nominal supply air flow rate";
+  Real epsSenPL
+    "Part load (75%) sensible heat exchanger effectiveness used for calculation";
+  Real epsSen_nominal
+    "Nominal sensible heat exchanger effectiveness used for calculation";
+  Real epsLatPL
+    "Part load (75%) latent heat exchanger effectiveness used for calculation";
+  Real epsLat_nominal
+    "Nominal latent heat exchanger effectiveness used for calculation";
+
+equation
+  // Check if the air flows are too unbalanced
+  assert(noEvent(VSup_flow - 2*VExh_flow < 0) or noEvent(VExh_flow - 2*VSup_flow < 0),
+    "*** Warning in " + getInstanceName() + ": The ratio of the supply flow rate to the exhaust flow rate should be in the range of [0.5, 2].",
+    level=AssertionLevel.warning);
+  // Calculate the average volumetric air flow and flow rate ratio.
+  rat = (VSup_flow + VExh_flow)/2/VSup_flow_nominal;
+  // Check if the extrapolation goes too far
+  assert(noEvent(rat > 0.5) and noEvent(rat < 1.3),
+    "*** Warning in " + getInstanceName() + ": The ratio of the operating flow rate to the nominal supply flow rate should be in the range of [0.5, 1.3].",
```

Comment:
We never write `*** Warning in`, just use `In `. The warning is written by the assertion.